aspnetcore/mvc/extensibility/app-parts.md

```diff
@@ -0,0 +1,122 @@
+
+If you have an assembly in your MVC project's dependency tree that has controllers in it you do not want to be used by the app, remove it from the `ApplicationPartManager`:
+
+<!-- literal_block {"ids": [], "linenos": true, "xml:space": "preserve", "language": "csharp"} -->
+
+```csharp
+services.AddMvc()
+    .ConfigureApplicationPartManager(p =>
+    {
+        var dependentLibrary = p.ApplicationParts
+            .FirstOrDefault(part => part.Name == "DependentLibrary");
+        if (dependentLibrary != null)
+        {
+           p.ApplicationParts.Remove(dependentLibrary);
+        }
+    })
+```
+
+In addition to your project's assembly and its dependent assemblies, the `ApplicationPartManager` will include parts for `Microsoft.AspNetCore.Mvc.TagHelpers` and `Microsoft.AspNetCore.Mvc.Razor` by default.
+
+## Application Feature Providers
+
+Application Feature Providers examine application parts and provide features for those parts. There are built-in feature providers for the following MVC features:
+
+- [Controllers](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.controllers.controllerfeatureprovider)
+- [Metadata Reference](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.razor.compilation.metadatareferencefeatureprovider)
+- [Tag Helpers](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.razor.taghelpers.taghelperfeatureprovider)
+- [View Components](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.viewcomponents.viewcomponentfeatureprovider)
+
+> [!NOTE]
+> Views do not use feature providers but will do so in a future release.
+
+Feature providers inherit from `IApplicationFeatureProvider<T>`, where `T` is the type of the feature. You can implement your own feature providers for any of MVC's feature types listed above. The order of feature providers in the `ApplicationPartManager.FeatureProviders` collection can be important, since later providers can react to actions taken by previous providers.
+
+### Sample: Generic Controller Feature
+
+Normally, ASP.NET Core MVC ignores generic controllers (for example, `SomeController<T>`). This sample uses a controller feature provider that runs after the default provider and adds generic controller instances for certain known types (defined in `EntityTypes.Types`):
+
+[!code-csharp[Main](./app-parts/sample/src/AppPartSample/GenericControllerFeatureProvider.cs?highlight=13&range=18-36)]
+
+The entity types:
+
+[!code-csharp[Main](./app-parts/sample/src/AppPartSample/Model/EntityTypes.cs?range=6-16)]
+
+The feature provider is added in `Startup`:
+
+<!-- literal_block {"ids": [], "linenos": true, "xml:space": "preserve", "language": "csharp"} -->
+
+```csharp
+services.AddMvc()
+        .ConfigureApplicationPartManager(p => p.FeatureProviders.Add(new GenericControllerFeatureProvider()));
```

Comment:
format to remove horiz sb (using the largest width browser window that has no TOC on the left or right - per my previous pic)